apis/meta/reference_types.go

```diff
@@ -54,3 +54,15 @@ type NamespacedObjectKindReference struct {
 	// +optional
 	Namespace string `json:"namespace,omitempty"`
 }
+
+// SecretKeyReference contains enough information to locate the referenced Kubernetes Secret object in the same
+// namespace. Optionally a key can be specified.
```

Comment:
```suggestion
// SecretKeyReference contains enough information to locate the referenced Kubernetes Secret object in the same
// namespace. Optionally, a key can be specified.
```

It might also be good for (developer) consumers to know in what way this type differs from `SecretKeySelector`, and when to pick this type over the version from K8s core.

Reply:
Would you like to add that information to the documentation above the type? Would like make its way into user-facing docs?

Reply:
Think this might already be the case for some of the types in this module. I am also happy with it being added to the `doc.go` file, or a dedicated doc block at the top of this document.

Reply:
Added above the type to keep it local.